modules/nf-scil/reconst/dtimetrics/main.nf

```diff
@@ -0,0 +1,119 @@
+        tuple val(meta), path("*__md.nii.gz")                      , emit: md, optional: true
+        tuple val(meta), path("*__mode.nii.gz")                    , emit: mode, optional: true
+        tuple val(meta), path("*__norm.nii.gz")                    , emit: norm, optional: true
+        tuple val(meta), path("*__rd.nii.gz")                      , emit: rd, optional: true
+        tuple val(meta), path("*__tensor.nii.gz")                  , emit: tensor, optional: true
+        tuple val(meta), path("*__nonphysical.nii.gz")             , emit: nonphysical, optional: true
+        tuple val(meta), path("*__pulsation_std_dwi.nii.gz")       , emit: pulsation_std_dwi, optional: true
+        tuple val(meta), path("*__residual.nii.gz")                , emit: residual, optional: true
+        tuple val(meta), path("*__residual_iqr_residuals.npy")     , emit: residual_iqr_residuals, optional: true
+        tuple val(meta), path("*__residual_mean_residuals.npy")    , emit: residual_mean_residuals, optional: true
+        tuple val(meta), path("*__residual_q1_residuals.npy")      , emit: residual_q1_residuals, optional: true
+        tuple val(meta), path("*__residual_q3_residuals.npy")      , emit: residual_q3_residuals, optional: true
+        tuple val(meta), path("*__residual_residuals_stats.png")   , emit: residual_residuals_stats, optional: true
+        tuple val(meta), path("*__residual_std_residuals.npy")     , emit: residual_std_residuals, optional: true
+        path "versions.yml"                                        , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    def mask =[]
+
+    if (b0mask) mask += ["--mask $b0mask"]
+    if ( task.ext.ad ) args += " --ad ${prefix}__ad.nii.gz"
+    if ( task.ext.evecs ) args += " --evecs ${prefix}__evecs.nii.gz"
+    if ( task.ext.evals ) args += " --evals ${prefix}__evals.nii.gz"
+    if ( task.ext.fa ) args += " --fa ${prefix}__fa.nii.gz"
+    if ( task.ext.ga ) args += " --ga ${prefix}__ga.nii.gz"
+    if ( task.ext.rgb ) args += " --rgb ${prefix}__rgb.nii.gz"
+    if ( task.ext.md ) args += " --md ${prefix}__md.nii.gz"
+    if ( task.ext.mode ) args += " --mode ${prefix}__mode.nii.gz"
+    if ( task.ext.norm ) args += " --norm ${prefix}__norm.nii.gz"
+    if ( task.ext.rd ) args += " --rd ${prefix}__rd.nii.gz"
+    if ( task.ext.tensor ) args += " --tensor ${prefix}__tensor.nii.gz"
+    if ( task.ext.nonphysical ) args += " --non-physical ${prefix}__nonphysical.nii.gz"
+    if ( task.ext.pulsation ) args += " --pulsation ${prefix}__pulsation_std_dwi.nii.gz"
+    if ( task.ext.residual ) args += " --residual ${prefix}__residual.nii.gz"
+
+
+    """
+    export ITK_GLOBAL_DEFAULT_NUMBER_OF_THREADS=1
+    export OMP_NUM_THREADS=1
+    export OPENBLAS_NUM_THREADS=1
+
+    scil_compute_dti_metrics.py $dwi $bval $bvec ${mask.join(" ")} $args -f --force_b0_threshold
```

Comment:
I think you should add the ``--not_all`` flag here since the script outputs all metrics by default. Currently, your metric flags would only change the filename and not make the metric computation conditional to the flag.

Reply:
What happens if the `--not_all` flag is omitted, but some of the metrics names are supplied ? Does it take the names, overwrite them or the script bugs ?

Reply:
From my understanding when I went through the script, if ``--not_all`` is not supplied but some of the metrics are, it simply changes the output name. I guess we could always supply the ``--not_all`` in the process, as @ThoumyreStanislas did in the last commit. Does that makes sense ?

Reply:
yes we've seen that with @AlexVCaron , we'll keep --not_all true all the time and the arguments we put in will be like the choice of output we want.